medicines/api/src/azure_search.rs

```diff
@@ -0,0 +1,84 @@
+use serde_derive::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct AzureHighlight {
+    #[serde(rename="content")]
+    content: Vec<String>
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureResult {
+    pub doc_type: String,
+    pub file_name: String,
+    pub metadata_storage_name: String,
+    pub metadata_storage_path: String,
+    pub product_name: Option<String>,
+    pub substance_name: Vec<String>,
+    pub title: String,
+    created: String,
+    facets: Vec<String>,
+    keywords: Option<String>,
+    metadata_storage_size: i32,
+    release_state: String,
+    rev_label: Option<String>,
+    suggestions: Vec<String>,
+    #[serde(rename="@search.score")]
+    score: f32,
+    #[serde(rename="@search.highlights")]
+    highlights: AzureHighlight,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AzureSearchResults {
+    pub value: Vec<AzureResult>,
+    #[serde(rename="@odata.context")]
+    context: String,
+    #[serde(rename="@odata.count")]
+    count: Option<i32>,
+}
+
+fn get_env(key: &str) -> String {
+    match std::env::var(key) {
+        Ok(b) => b,
+        Err(_) => "".to_owned()
+    }
+}
```

Comment:
So this swallows errors? Should it be named `get_env_or_empty_string` to make its purpose more clear?

Reply:
Worth discussing. If we do let the error bubble up, it's swallowed. We may need to move this so that if the envrionment variables are missing, the server doesn't start, so we'll see healthchecks fail.

Reply:
I think we should handle errors explicitly. An environment variable not being set is usually going to result in an un-usable API so we may as well fatal here.

Reply:
@craiga @m-doughty This now `panic`s when necessary environmental variables are not found. I also rearchitected so that this is called when building a global context, so starting up the server will fail quickly and loudly.